camel/toolkits/ask_news_toolkit.py

```diff
@@ -0,0 +1,380 @@
+        else:
+            raise ValueError(f"Invalid return_type: {return_type}")
+
+    def get_news(
+        self,
+        query: str,
+        n_articles: int = 10,
+        return_type: Literal["string", "dicts", "both"] = "string",
+        method: Literal["nl", "kw"] = "kw",
+    ) -> Union[str, dict, Tuple[str, dict]]:
+        r"""Fetch news or stories based on a user query.
+
+        Args:
+            query (str): The search query for fetching relevant news or
+            stories.
+            n_articles (int): Number of articles to include in the response.
+                (default: :obj:`10`)
+            return_type (Literal["string", "dicts", "both"]): The format of the
+                return value. (default: :obj:`"string"`)
+            method (Literal["nl", "kw"]): The search method, either "nl" for
+                natural language or "kw" for keyword search. (default:
+                :obj:`"kw"`)
+
+        Returns:
+            Union[str, dict, Tuple[str, dict]]: A string, dictionary, or both
+            containing the news or story content, or an error message if the
+            process fails.
+        """
+        try:
+            response = self.asknews_client.news.search_news(
+                query=query,
+                n_articles=n_articles,
+                return_type=return_type,
+                method=method,
+            )
+
+            return self._process_response(response, return_type)
+
+        except Exception as e:
+            raise Exception(
+                f"An error occurred while fetching news for '{query}': {e!s}."
+            )
+
+    def search_reddit(
+        self,
+        keywords: list,
+        return_type: Literal["string", "dicts", "both"] = "string",
+    ) -> Union[str, dict, Tuple[str, dict]]:
+        r"""Search Reddit based on the provided keywords.
+
+        Args:
+            keywords (list): The keywords to search for on Reddit.
+            return_type (Literal["string", "dicts", "both"]): The format of the
+                return value. (default: :obj:`"string"`)
+
+        Returns:
+            Union[str, dict, Tuple[str, dict]]: The Reddit search results as a
+            string, dictionary, or both.
+
+        Raises:
+            Exception: If there is an error while searching Reddit.
+        """
+        try:
+            response = self.asknews_client.news.search_reddit(
+                keywords=keywords
+            )
+
+            return self._process_response(response, return_type)
+
+        except Exception as e:
+            raise Exception(
+                f"An error occurred while searching Reddit: {e!s}."
+            )
+
+    def get_stories(
+        self,
+        categories: list,
+        continent: str,
+        sort_by: str = "coverage",
+        sort_type: str = "desc",
+        reddit: int = 3,
+        expand_updates: bool = True,
+        max_updates: int = 2,
+        max_articles: int = 10,
```

Comment:
as we are going to provide this tool to agent, I feel some unimportant argument can be excluded like `sort_type`